analysis/analysis.py

```diff
@@ -356,9 +591,81 @@ def setUp(self):
                 'scroll_ups': 0,
             }
         ]
+        test_data_2_path = Path('data', 'test_data_2.csv')
+        self.default_student_data_2 = load_data_csv(test_data_2_path)
         sample_csv_path = Path('data', 'rereading_data_2019-09-13.csv')
         self.student_data = load_data_csv(sample_csv_path)
 
+    def test_mean_rereading_time_for_a_question(self):
+        """
+        Tests mean_rereading_time_for_a_question function with many data sets and checks if
+        the function crashes when it encounters the default data set. Also test many cases with
+        all question and context combinations.
+        """
+        mean_rereading_data = mean_rereading_time_for_a_question(self.default_student_data, "", "")
+
+        empty_comparison_tuple = ("", "", 0, 0)
+        self.assertEqual(mean_rereading_data, empty_comparison_tuple)
+
+        feel = "In one word, how does this text make you feel?"
+        about = "In three words or fewer, what is this text about?"
+        encountered = "Have you encountered this text before?"
+        ads = "This is an ad."
+        short_story = "This is actually a short story."
+
+        mean_rereading_time_results_data = [
+            mean_rereading_time_for_a_question(self.test_student_data, feel, ads),
+            mean_rereading_time_for_a_question(self.test_student_data, about, ads),
+            mean_rereading_time_for_a_question(self.test_student_data, encountered, ads),
+            mean_rereading_time_for_a_question(self.test_student_data, feel, short_story),
+            mean_rereading_time_for_a_question(self.test_student_data, about, short_story),
+            mean_rereading_time_for_a_question(self.test_student_data, encountered, short_story)
+        ]
+
+        # The expected result times are rounded to 2 decimals here due to Python rounding errors
+        # not matching actual rounding.
+        mean_comparison_results = [
+            (feel, ads, round(2.319, 2), 1),
+            (about, ads, round(2.945, 2), 1),
+            (encountered, ads, 0, 0),
+            (feel, short_story, round(1.121, 2), 1),
+            (about, short_story, 0, 0),
+            (encountered, short_story, 0, 0)
+        ]
+        self.assertEqual(mean_rereading_time_results_data, mean_comparison_results)
+
+    def test_mean_rereading_time_for_a_question_two(self):
```

Comment:
It's not really clear to me that this is testing something not covered by the above test case.

Reply:
Before, our method to find the mean rereading time found the question based on a unique keyword. Myke pointed that there could be errors in that so we were working with a test to see what the issues of referencing to question by keyword could be. Should we remove this test method now?

Reply:
yep - it's now redundant with the other test.